### PR TITLE
Add operational reserves margins

### LINKED ISSUE #321: add operational reserve analog to PyPSA-EUR and GenX
See implementation here: https://github.com/PyPSA/pypsa-eur/commit/3678e5c523080190468e2ce235663df43e644714

Add additional a switch as done in: https://github.com/PyPSA/pypsa-eur/commit/84e146834c412edb6f402255dd1fcdcc77de3cd6

Fix operational margin to work with extendible generators: https://github.com/PyPSA/pypsa-eur/commit/630fb9783f5acb9586f28d9732cc97ecad288524

### PR DESCRIPTION
**Pull request by pz-max**
## Changes proposed in this Pull Request
This PR adds operational reserve margins from PyPSA-Eur as described here: https://github.com/pypsa-meets-africa/pypsa-africa/issues/395

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by pz-max** [comment]
@fneum @FabianHofmann thanks for the awesome work. Just to track some changes compared to your changes:
- I add configtables for the new options
- I set the default of contingency to **0** as the contingency is highly dependent on the region of interest e.g. 4000 MW contingency is too high for modelling Croatia only.
- I activate the operational reserve margin in the CI run so that it will be tested

Maybe some of it can be reused in PyPSA-Eur?

**Comment by pz-max** [comment]
@davide-f I changed the main branch to be protected, meaning we need at least one review before merging.

File: config.default.yaml [hunk 1]
```diff
@@ -93,6 +92,12 @@ electricity:
   co2base: 1.487e9  # European default, adjustment to Africa necessary
   agg_p_nom_limits: data/agg_p_nom_minmax.csv
 
+  operational_reserve: # like https://genxproject.github.io/GenX/dev/core/#Reserves
+    activate: false
+    epsilon_load: 0.02 # share of total load
+    epsilon_vres: 0.02 # share of total renewable supply
+    contingency: 0 # fixed capacity in MW
```
Comment (davide-f) [inline config.default.yaml:99]: Ok for the time being, in the future, contingency may be calculated as a fraction of the installed capacity plus a security factor

**Review by davide-f — APPROVED** [review]
Great job! :D

File: config.default.yaml (hunk 1 above)
Comment (pz-max) [reply config.default.yaml:99]: Let's see if it is necessary in future. People can add it on demand.

**Pull request merged**